packages/nextjs/src/utils/withSentry.ts

```diff
@@ -74,7 +74,57 @@ export const withSentry = (origHandler: NextApiHandler): WrappedNextApiHandler =
       }
 
       try {
-        return await origHandler(req, res);
+        const handlerResult = await origHandler(req, res);
+
+        // Temporarily mark the response as finished, as a hack to get nextjs to not complain that we're coming back
+        // from the handler successfully without `res.end()` having completed its work.  This is necessary (and we know
+        // we can do it safely) for a few reasons:
+        //
+        // - Normally, `res.end()` is sync and completes before the request handler returns, as part of the handler
+        //   sending data back to the client. As soon as the handler is done, nextjs checks to make sure that the
+        //   response is indeed finished. (`res.end()` signals this by setting `res.finished` to `true`.) If it isn't,
+        //   nextjs complains.("Warning: API resolved without sending a response for <url>.")
```

Comment:
```suggestion
        //   nextjs complains. ("Warning: API resolved without sending a response for <url>.")
```

And we should link to https://github.com/getsentry/sentry-javascript/issues/3852.

Reply:
Good catch on the proofreading. As for linking, I feel like if we go there, we could end up with issue links all over the code. It's easy enough to do `git blame` and come look at the PR, which has the link.